temporalio/worker/_workflow_instance.py

```diff
@@ -168,6 +168,23 @@ def activate(
         """
         raise NotImplementedError
 
+    @abstractmethod
+    def get_payload_codec_with_context(
```

Comment:
Not sure sandboxed instance code should concern itself with codecs. Pedantic, but I think a cleaner separation is probably `get_pending_serialization_context(pending_type: SomeEnumSayingWhichType, seq: int) -> Optional[BaseWorkflowSerializationContext]`.